dapps/profile.vue/src/components/profile/company/registration/registration.ts

```diff
@@ -0,0 +1,145 @@
+  court: EvanFormControl;
+  register: EvanFormControl;
+  registerNumber: EvanFormControl;
+  salesTaxID: EvanFormControl;
+}
+
+@Component({ })
+export default class CompanyRegistrationForm extends mixins(EvanComponent) {
+  /**
+   * Address for that the data should be loaded
+   */
+  @Prop() address;
+
+  /**
+   * Evan form instance for registration data.
+   */
+  registrationForm: RegistrationFormInterface = null;
+
+  /**
+   * Watch for dispatcher updates
+   */
+  listeners: Array<any> = [ ];
+
+  /**
+   * Load the mail details
+   */
+  async created() {
+    // watch for save updates
+    this.listeners.push(dispatchers.updateProfileDispatcher.watch(($event: any) => {
+      if ($event.detail.status === 'finished' || $event.detail.status === 'deleted') {
+        this.loadProfileData();
+      }
+    }));
+
+    // load profile data
+    await this.loadProfileData();
+  }
+
+  /**
+   * Clear dispatcher listeners
+   */
+  beforeDestroy() {
+    this.listeners.forEach(listener => listener());
+  }
+
+  /**
+   * Load the profile data an specify the registration form.
+   */
+  async loadProfileData() {
+    const runtime = (<any>this).getRuntime();
+    const registrationData = await ProfileMigrationLibrary.loadProfileData(runtime, 'registration');
+
+    // setup registration form
+    this.registrationForm = (<RegistrationFormInterface>new EvanForm(this, {
+      company: {
+        value: registrationData.company || '',
+        validate: function(vueInstance: CompanyRegistrationForm, form: RegistrationFormInterface) {
+          return this.value.length !== 0;
+        },
+      },
+      court: {
+        value: registrationData.court || '',
+        validate: function(vueInstance: CompanyRegistrationForm, form: RegistrationFormInterface) {
+          return this.value.length !== 0;
+        },
+      },
+      register: {
+        value: registrationData.register || '',
+        validate: function(vueInstance: CompanyRegistrationForm, form: RegistrationFormInterface) {
+          return this.value.length !== 0;
+        },
+        uiSpecs: {
+          type: 'select',
+          attr: {
+            options: [
+              { value: 'germany', label: '_profile.company.registration.countries.germany', }
+            ],
+          }
+        }
+      },
+      registerNumber: {
+        value: registrationData.registerNumber || '',
+        validate: function(vueInstance: CompanyRegistrationForm, form: RegistrationFormInterface) {
+          return this.value.length !== 0;
+        },
+      },
+      salesTaxID: {
+        value: registrationData.salesTaxID || '',
+        validate: function(vueInstance: CompanyRegistrationForm, form: RegistrationFormInterface) {
+          return this.value.length !== 0;
+        },
+      },
+    }));
+  }
+
+  async changeProfileData() {
+    dispatchers.updateProfileDispatcher.start((<any>this).getRuntime(), {
+      formData: this.registrationForm.toObject(),
```

Comment:
what's doeas `toObject()` ?

Reply:
It's a function that maps all controls with it's names as keys to the current control values as values into one object.